cosmos.v1.proto

```diff
@@ -0,0 +1,482 @@
+syntax = "proto3";
+
+package cosmos.v1;
+
+option go_package = "github.com/figment-networks/tendermint-protobuf-def/cosmos";
+
+import "codec.proto";
+import "gogo.proto";
+import "google/protobuf/descriptor.proto";
+
+option(gogoproto.goproto_stringer_all) = false;
+option(gogoproto.stringer_all)         = false;
+
+enum SignMode {
+  SIGN_MODE_UNSPECIFIED = 0; SIGN_MODE_DIRECT = 1; SIGN_MODE_TEXTUAL = 2; SIGN_MODE_DIRECT_AUX = 3;
+  SIGN_MODE_LEGACY_AMINO_JSON                                                                  = 127;
+}
+
+message Tx {
+  TxBody         body       = 1;
+  AuthInfo       auth_info  = 2;
+  repeated bytes signatures = 3;
+}
+
+message TxBody {
+  repeated Any messages                       = 1;
+  string       memo                           = 2;
+  uint64       timeout_height                 = 3;
+  repeated Any extension_options              = 1023;
+  repeated Any non_critical_extension_options = 2047;
+}
+
+message AuthInfo {
+  repeated SignerInfo signer_infos = 1;
+  Fee                 fee          = 2;
+  Tip                 tip          = 3;
+}
+
+message SignerInfo {
+  Any      public_key = 1;
+  ModeInfo mode_info  = 2;
+  uint64   sequence   = 3;
+}
+
+message ModeInfo {
+  oneof sum {
+    Single single = 1;
+    Multi  multi  = 2;
+  }
+}
+
+message Single { SignMode mode = 1; }
+
+message Multi {
+  CompactBitArray   bitarray   = 1;
+  repeated ModeInfo mode_infos = 2;
+}
+
+// cosmos.crypto.multisig.v1beta1.CompactBitArray
+message CompactBitArray {
+  option(gogoproto.goproto_stringer) = false;
+
+  uint32 extra_bits_stored = 1;
+  bytes  elems             = 2;
+}
+
+message Fee {
+  repeated Coin amount =
+      1 [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
```

Comment:
is the castrepeated something we added? If so, what is it needed for?

Reply:
it's copied from cosmos-sdk, should I update castrepeated value?